plugins/aws-appmesh/config/netconfig.go

```diff
@@ -142,6 +142,20 @@ func validateConfig(config netConfigJSON) error {
 		return fmt.Errorf("missing parameter appPorts (required if proxyIngressPort is provided)")
 	}
 
+	// For configs coming from appmesh controller, the IngressPort will always be set but the appPort can be empty string with single element
+	// We still treat that as valid and delete that empty element
+	if len(config.AppPorts) == 1 && config.AppPorts[0] == "" {
```

Comment:
Why doesn't the check in line 167 cover this case?

Reply:
This is just fixing the symptom, not the root cause. Why is an empty port sent to the CNI plugin in the first place? CNI plugin can be defensive, but you need to figure out what made the controller send an empty port string.

Reply:
The line in 167 does not change the value, only skipping the empty ones. We will need to work with fargate team to figure out why this empty string is passed to cni. My assumption is annotation is not checked and set to null before going to netconfig. So it remains an empty string.